File: backend-api/src/main/java/com/smartenergy/backendapi/model/HorarioAcademico.java
package com.smartenergy.backendapi.model;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalTime;

@Getter
@Setter
@Entity
@Table(name = "HorarioAcademico")
public class HorarioAcademico extends BaseEntity {
    private String asignatura;
    private int diaSemana;
    private LocalTime horaFin;
    private LocalTime horaInicio;

    @ManyToOne
    @JoinColumn(name = "aula_id")
    private Aula aula;
}

File: mobile-app/app/src/main/java/com/example/smartenergy/model/HorarioAcademico.kt
package com.example.smartenergy.model

import java.time.LocalTime
import java.util.UUID

data class HorarioAcademico(
    val id: String = UUID.randomUUID().toString(),
    val asignatura: String,
    val diaSemana: Int,
    val horaInicio: LocalTime,
    val horaFin: LocalTime,
    val aula: Aula? = null
)
